RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -2188,8 +2188,36 @@ boolean haveAnyIotmAlternativeRestSiteAvailable(){
  */
 boolean doFreeRest(){
 	if(haveFreeRestAvailable()){
+
+		// burn MP if possible prior to resting
+		int restorableMp = my_maxmp() - my_mp();
+		int mpToBurn = 0;
+		if (chateaumantegna_available() || auto_campawayAvailable()) // will restore at least 100 MP
+		{
+			mpToBurn = 100 - restorableMp;
+		} else if (get_dwelling() == $item[Frobozz Real-Estate Company Instant House (TM)])
+		{
+			mpToBurn = 40 - restorableMp;
+		} else if (get_dwelling() == $item[Newbiesport&trade; tent])
+		{
+			mpToBurn = 10 - restorableMp;
+		} else if (get_dwelling() == $item[big rock]) // just the ground
```

Comment:
Remove the if here so it is just an else. Having a default will make it more robust. Add comment explaining default assumes resting in dirt

Reply:
Done